cub/cub/device/dispatch/tuning/tuning_reduce.cuh

```diff
@@ -279,6 +372,94 @@ struct policy_hub
 
   using MaxPolicy = Policy1000;
 };
+
+struct arch_policies // equivalent to the policy_hub, holds policies for a bunch of CUDA architectures
+{
+  accum_type at;
+  op_type ot;
+  offset_size os;
+  int accum_size;
```

Comment:
I'm a little bit worried about this. The way we have constructed it here, it is an ABI (and possibly API in some cases?) break to add "traits" to the selection process; same for exposing more knobs to the user. I wonder if instead of doing this, having a user / the dispatch code actually pass in something more resembling an environment that provides the queries. Yes, that'd turn this back into a template, but if we consistently passed in an equivalently-typed environment, that wouldn't be too bad, *and* it'd provide us ways of extending the knobs without breaks in the future.